databricks/koalas/frame.py

```diff
@@ -2897,6 +2897,157 @@ def merge(left, right: 'DataFrame', how: str = 'inner',
 
         return DataFrame(joined_table.select(*exprs))
 
+    # TODO: implenment `on` parameter. See Pandas's DataFrame.join.
+    def join(left, other: 'DataFrame',
+             on: Optional[Union[str, List[str]]] = None, how: str = 'left',
+             lsuffix: str = '', rsuffix: str = '', sort: bool = False) -> 'DataFrame':
+        """
+        Join columns of another DataFrame.
+
+        Join columns with other DataFrame either on index or on a key column. Efficiently join
+        multiple DataFrame objects by index at once by passing a list.
+
+        Parameters
+        ----------
+        other: DataFrame, Series, or list of DataFrame.
+        on: str, list of str, or array-like, optional
+            Column or index level name(s) in the caller to join on the index in other, otherwise
+            joins index-on-index. If multiple values given, the other DataFrame must have a
+            MultiIndex. Can pass an array as the join key if it is not already contained in the
+            calling DataFrame. Like an Excel VLOOKUP operation.
+        how: How to handle the operation of the two objects.
+            {‘left’, ‘right’, ‘outer’, ‘inner’}, default‘inner’
+            left: use calling frame’s index (or column if on is specified).
+            right: use other’s index.
+            outer: form union of calling frame’s index (or column if on is specified) with
+            other’s index, and sort it. lexicographically.
+            inner: form intersection of calling frame’s index (or column if on is specified)
+            with other’s index, preserving the order of the calling’s one.
+
+        Returns
+        -------
+        DataFrame
+            A dataframe containing columns from both the caller and other.
+
+        See Also
+        --------
+        DataFrame.merge: For column(s)-on-columns(s) operations.
+
+        Notes
+        -----
+        Parameters on, lsuffix, and rsuffix are not supported when passing a list of DataFrame
+        objects.
+
+        Examples
+        --------
+        >>> kdf1 = ks.DataFrame({'key': ['K0', 'K1', 'K2', 'K3', 'K4', 'K5'],
```

Comment:
Couldn't the example DataFrames be a bit shorter, so that they're easier to read while showcasing the same functionality?